tool/tsh/putty_config_windows.go

```diff
@@ -0,0 +1,384 @@
+	if err != nil {
+		// ERROR_FILE_NOT_FOUND is an acceptable error, meaning that the value does not already
+		// exist and it must be created
+		if err != syscall.ERROR_FILE_NOT_FOUND {
+			log.Debugf("Can't get registry value %v: %T", registryKeyName, err)
+			return false, trace.Wrap(err)
+		}
+	} else {
+		// initialise empty hostlist if no value returned
+		if len(hostList) == 0 {
+			hostList = []string{}
+		}
+	}
+	defer registryKey.Close()
+
+	// iterate over the list of hostnames provided
+	// if an FQDN is provided and there are already entries in the list, see whether it can be covered
+	// by a wildcard hostname that already exists in the list and skip adding it.
+	if len(hostList) > 0 && strings.Contains(hostname, ".") {
+		fullHostname := strings.Split(hostname, ".")
+		wildcardDomain := fmt.Sprintf("*.%s", strings.Join(fullHostname[1:], "."))
+		if !slices.Contains(hostList, wildcardDomain) {
+			log.Debugf("Adding wildcard %q to hostList", wildcardDomain)
+			hostList = append(hostList, wildcardDomain)
+			log.Debugf("Removing hostname %q from hostList as it's now covered by a wildcard", hostname)
+			hostList = utils.RemoveFromSlice(hostList, hostname)
+		} else {
+			log.Debugf("Not adding %q because it's already covered by %q", hostname, wildcardDomain)
+		}
+	} else {
+		if !slices.Contains(hostList, hostname) {
+			log.Debugf("Adding %q to hostList", hostname)
+			hostList = append(hostList, hostname)
+		} else {
+			log.Debugf("%q is already present in hostList", hostname)
+		}
+	}
+	sort.Strings(hostList)
+
+	// write strings to subkey
+	if ok, err := registryWriteMultiString(registryKey, "MatchHosts", hostList); !ok {
+		return false, trace.Wrap(err)
+	}
+	if ok, err := registryWriteString(registryKey, "PublicKey", publicKey); !ok {
+		return false, trace.Wrap(err)
+	}
+
+	// write dwords for signature acceptance
+	if ok, err := registryWriteDword(registryKey, "PermitRSASHA1", puttyPermitRSASHA1); !ok {
+		return false, trace.Wrap(err)
+	}
+	if ok, err := registryWriteDword(registryKey, "PermitRSASHA256", puttyPermitRSASHA256); !ok {
+		return false, trace.Wrap(err)
+	}
+	if ok, err := registryWriteDword(registryKey, "PermitRSASHA512", puttyPermitRSASHA512); !ok {
+		return false, trace.Wrap(err)
+	}
+
+	return true, nil
+}
+
+// formatLocalCommandString replaces placeholders in a constant with actual values
+func formatLocalCommandString(tshPath string, cluster string) string {
+	// replace the placeholder "%cluster" with the actual cluster name as passed to the function
+	clusterString := strings.ReplaceAll(puttyProxyTelnetCommand, "%cluster", cluster)
+	// PuTTY needs its paths to be double-escaped i.e. C:\\Users\\User\\tsh.exe
+	escapedTshPath := strings.ReplaceAll(tshPath, `\`, `\\`)
+	return strings.ReplaceAll(clusterString, "%tsh", escapedTshPath)
+}
+
+// onPuttyConfig handles the `tsh config putty` subcommand
+func onPuttyConfig(cf *CLIConf) error {
+	if runtime.GOOS != constants.WindowsOS {
```

Comment:
I think you can remove this check since the `_windows.go` suffix means this file will only be built for Windows.